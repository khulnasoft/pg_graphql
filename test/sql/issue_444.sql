begin;

    create function "someFunc" (arg uuid)
        returns int
        immutable
        language sql
    as $$ select 1; $$;

    select jsonb_pretty(
        graphql.resolve($$
            {
            __type(name: "Query") {
                fields(includeDeprecated: true) {
                    name
                    args {
                      name
                      type {
                        kind
                        name
                        ofType {
                          kind
                          name
                        }
                      }
                    }
                }

            }
            }
        $$)
    );

rollback;
