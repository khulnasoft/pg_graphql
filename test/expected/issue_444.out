begin;
    create function "someFunc" (arg uuid)
        returns int
        immutable
        language sql
    as $$ select 1; $$;
    select jsonb_pretty(
        graphql.resolve($$
            {
            __type(name: "Query") {
                fields(includeDeprecated: true) {
                    name
                    args {
                      name
                      type {
                        kind
                        name
                        ofType {
                          kind
                          name
                        }
                      }
                    }
                }

            }
            }
        $$)
    );
                     jsonb_pretty                      
-------------------------------------------------------
 {                                                    +
     "data": {                                        +
         "__type": {                                  +
             "fields": [                              +
                 {                                    +
                     "args": [                        +
                         {                            +
                             "name": "nodeId",        +
                             "type": {                +
                                 "kind": "NON_NULL",  +
                                 "name": null,        +
                                 "ofType": {          +
                                     "kind": "SCALAR",+
                                     "name": "ID"     +
                                 }                    +
                             }                        +
                         }                            +
                     ],                               +
                     "name": "node"                   +
                 },                                   +
                 {                                    +
                     "args": [                        +
                         {                            +
                             "name": "arg",           +
                             "type": {                +
                                 "kind": "NON_NULL",  +
                                 "name": null,        +
                                 "ofType": {          +
                                     "kind": "SCALAR",+
                                     "name": "UUID"   +
                                 }                    +
                             }                        +
                         }                            +
                     ],                               +
                     "name": "someFunc"               +
                 }                                    +
             ]                                        +
         }                                            +
     }                                                +
 }
(1 row)

rollback;
